src/test/java/com/exasol/adapter/capabilities/CapabilitiesTest.java

```diff
@@ -18,67 +21,116 @@ void beforeEach() {
-                () -> assertEmptyScalarFunctionCapabilities(capabilities), //
-                () -> assertEmptyAggregateFunctionCapatilities(capabilities));
+              () -> assertEmptyLiteralCapabilities(capabilities), //
+              () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities), //
+              () -> assertEmptyScalarFunctionCapabilities(capabilities), //
+              () -> assertEmptyAggregateFunctionCapatilities(capabilities));
     }
 
     @Test
     void buildWithScalarFunctionCapabilities() {
-        final ScalarFunctionCapability[] expectedCapabilities = { ScalarFunctionCapability.ABS,
-                ScalarFunctionCapability.ACOS };
+        final ScalarFunctionCapability[] expectedCapabilities =
+              {ScalarFunctionCapability.ABS, ScalarFunctionCapability.ACOS};
         final Capabilities capabilities = this.builder.addScalarFunction(expectedCapabilities).build();
         assertAll(() -> assertEmptyMainCapabilities(capabilities), //
-                () -> assertEmptyLiteralCapabilities(capabilities), //
-                () -> assertEmptyPredicateCapabilities(capabilities), //
-                () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities), //
-                () -> assertEmptyAggregateFunctionCapatilities(capabilities));
+              () -> assertEmptyLiteralCapabilities(capabilities), //
+              () -> assertEmptyPredicateCapabilities(capabilities), //
+              () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities), //
+              () -> assertEmptyAggregateFunctionCapatilities(capabilities));
     }
 
     @Test
     void buildWithAggregateFunctionCapabilities() {
-        final AggregateFunctionCapability[] expectedCapabilities = {
-                AggregateFunctionCapability.APPROXIMATE_COUNT_DISTINCT, AggregateFunctionCapability.AVG };
+        final AggregateFunctionCapability[] expectedCapabilities =
+              {AggregateFunctionCapability.APPROXIMATE_COUNT_DISTINCT, AggregateFunctionCapability.AVG};
         final Capabilities capabilities = this.builder.addAggregateFunction(expectedCapabilities).build();
         assertAll(() -> assertEmptyMainCapabilities(capabilities), //
-                () -> assertEmptyLiteralCapabilities(capabilities), //
-                () -> assertEmptyPredicateCapabilities(capabilities), //
-                () -> assertEmptyScalarFunctionCapabilities(capabilities), //
-                () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities));
+              () -> assertEmptyLiteralCapabilities(capabilities), //
+              () -> assertEmptyPredicateCapabilities(capabilities), //
+              () -> assertEmptyScalarFunctionCapabilities(capabilities), //
+              () -> assertCapabilitesContainAllOf(capabilities, expectedCapabilities));
+    }
+
+    @Test
+    void substractCapabilities() {
```

Comment:
Method to long. Please split into five separate tests -- one for each capability group.